File: backend/src/main/java/com/awais/hr/module/performance/service/PerformanceService.java
package com.awais.hr.module.performance.service;

import com.awais.hr.module.performance.dto.GoalProgressUpdateDTO;
import java.util.List;
import java.util.Map;

public interface PerformanceService {
    List<Map<String, Object>> getGoals(String email);
    void updateGoalProgress(String id, GoalProgressUpdateDTO dto);
    void createGoal(String email, String title, int targetValue);
    void submitPeerFeedback(String email, String targetEmployeeId, String feedback, int rating);
    List<Map<String, Object>> getPeerFeedback(String email);
}

File: frontend/src/services/performanceService.js
import apiClient from './api';

export async function getGoals() {
  return apiClient.get('/suite/performance/goals');
}

export async function createGoal(goalData) {
  return apiClient.post('/suite/performance/goals', goalData);
}

export async function updateGoalProgress(id, progress) {
  return apiClient.put(`/suite/performance/goals/${id}`, { currentValue: progress });
}

export async function getPeerFeedback() {
  return apiClient.get('/suite/performance/peer-feedback');
}

export async function submitPeerFeedback(feedbackData) {
  return apiClient.post('/suite/performance/peer-feedback', feedbackData);
}
